sql-common/client.c

```diff
@@ -2885,6 +2954,21 @@ CLI_MYSQL_REAL_CONNECT(MYSQL *mysql,const char *host, const char *user,
         continue;
       }
 
+      if (bind_addr)
+      {
+        for (bres= bind_addr; bres; bres= bres->ai_next)
+        {
+          if (!(rc= bind(sock, bres->ai_addr, (int)bres->ai_addrlen)))
+            break;
+        }
+        if (rc)
+        {
+          closesocket(sock);
+          sock= INVALID_SOCKET;
```

Comment:
Does `saved_error` like the previous block need to happen here too?

Reply:
You are right. Fixed